core/src/main/scala/org/apache/spark/deploy/security/HiveCredentialProvider.scala

```diff
@@ -29,11 +29,10 @@ import org.apache.hadoop.io.Text
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
 import org.apache.hadoop.security.token.Token
 
-import org.apache.spark.SparkConf
 import org.apache.spark.internal.Logging
 import org.apache.spark.util.Utils
 
-private[security] class HiveCredentialProvider extends ServiceCredentialProvider with Logging {
+private[security] class HiveCredentialProvider extends HadoopDelegationTokenProvider with Logging {
```

Comment:
Is this the only reason we need to make spark core depend on hive? My concern is that, previously it's possible for users to build spark without hive, but now it's impossible.

Can we move this file to hive module and use reflection in `ConfigurableCredentialManager`? so that if users build spark without hive and wanna use `HiveCredentialProvider`, we throw an exception.